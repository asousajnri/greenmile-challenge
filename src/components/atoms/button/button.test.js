import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

import Button from './button';

describe('<Button />', () => {
  it('should render without props', () => {
    const { container } = render(<Button />);

    const buttonText = screen.getByText(/Button/i, {
      selector: 'span',
    });

    expect(buttonText).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render prop (Loading)', () => {
    const { container } = render(<Button isLoading disabled />);

    const buttonText = screen.getByText(/Carregando.../i, {
      selector: 'span',
    });

    expect(buttonText).toBeInTheDocument();
    expect(container.firstChild).toBeDisabled();
    expect(container.firstChild).toHaveStyle({
      pointerEvents: 'none',
    });
  });

  it('should render with icon', () => {
    render(<Button label="Button" icon="FiFeather" />);

    const ariaLabelIcon = screen.getByLabelText(/button-icon/i, {
      selector: 'svg',
    });

    expect(ariaLabelIcon).toBeInTheDocument();
  });

  it('should call onCLick when user clicks', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick} />);

    const button = screen.getByRole('button');
    userEvent.click(button);

    expect(onClick).toHaveBeenCalled();
  });

  it('should call onCLick once one', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick} />);

    const button = screen.getByRole('button');
    userEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
