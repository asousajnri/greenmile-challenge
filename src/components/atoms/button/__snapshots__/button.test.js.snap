// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render without props 1`] = `
<button
  class="sc-bdfBwQ hfpDAm"
>
  <span
    aria-label="button-text"
  >
    Button
  </span>
</button>
`;
